arduino/smartcar/smartcar.ino

```diff
@@ -1,9 +1,45 @@
-#include <Smartcar.h>
+#include <SmartCar.h>
+
+const int triggerPin = 6;
+const int echoPin = 7;
+const unsigned int maxDistance = 100;
+
+ArduinoRuntime arduinoRuntime;
+
+BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
+
+DifferentialControl control(leftMotor, rightMotor);
+
+SR04 frontSensor(arduinoRuntime, triggerPin, echoPin, maxDistance);
+
+SimpleCar car(control);
+
+const auto forwardSpeed = 30;
+const auto stopSpeed = 0;
+
+void stopCar(){
+    car.setSpeed(stopSpeed);
+}
+void startCar(){
+    car.setSpeed(forwardSpeed);
+}
 
 void setup() {
-  // put your setup code here, to run once:
+
+    Serial.begin(9600);
 }
 
 void loop() {
-  // put your main code here, to run repeatedly:
+
+    Serial.println(frontSensor.getDistance());
+    delay(100);
+    if(frontSensor.getDistance() <= 50 && frontSensor.getDistance() != 0){
```

Comment:
This is still problematic... What will happen if during the first call for getDistance 0 is returned and in the second 99?

Reply:
I don't really understand what you mean...the car will still go forward when the call is 99.

Reply:
No it won't. ;)

Read the scenario carefully. I am referring to line 37, don't care so much about line 35:

> first call for `getDistance` 0 is returned and in the second 99

What happens then?

Reply:
Oh I understand, so if the first call is 0 it means that it is less than 50, so that holds. In the second call if it is 99 then that also holds because it is not equal to 0, and thus, the whole if block executes. However it has to be within the 0 and 50 range. So the correct way would be to have less than or equal to 50 and greater than 0.

Reply:
> So the correct way would be to have less than or equal to 50 and greater than 0

You are correct about the problem, but not so much about the solution. Keep thinking... :hammer:

Reply:
Greater than 0 and less than 50, because equaling to 50 would mean that it will hit the obstacle, but if its smaller than the distance to the obstacle, it will never hit it.

Reply:
You are going further away from the point... :) 
> What will happen if during the first call for getDistance 0 is returned and in the second 99?

Instead of trying to "fix" the issue, you should instead _remove_ it. Don't know how else to phrase it without giving you the solution. :sweat_smile:

Reply:
Okay, so getDistance has to be assigned to a constant variable in order to avoid inconsistency in the if block?

Reply:
Yes, what you need to understand is that each `getDistance` call is a **new** measurement that could very well be different than the one right before it.

Reply:
That makes sense, it was not apparent to me before. Thank you for the help!